src/toil/lib/docker.py

```diff
@@ -222,19 +222,16 @@ def toil_job(job):
     client = docker.from_env(version='auto', timeout=timeout)
 
     if deferParam == STOP:
-        job.defer(dockerStop, containerName)
-
-    if deferParam == FORGO:
-        remove = False
+        job.defer(dockerStop, containerName, remove)
+    elif deferParam == FORGO:
```

Comment:
I hope I didn't change the behavior here... If I understand correctly, we had `if` here so it is possible to remove after stop if `deferParam == STOP`. But now I've moved the remove option to `dockerStop / dockerKill`.

Reply:
It looks like the original behavior maybe didn't match the docstring. `RM` is supposed to be the default if no `deferParam` is passed, but both now and before it looks like for a `deferParam` of `None`, no deferred cleanup is done (unless `remove` is set).

Reply:
> It looks like the original behavior maybe didn't match the docstring. 

Was there a reason for this? I don't want to make `RM` to be the default and break something else.

Reply:
If there's a reason for it to be the way it is now that was important enough to us that we wanted to remember it later, we would have written it down in the documentation, or a comment. Or *maybe* a commit message.

I think the reason the behavior doesn't match the documentation is just a mistake.